packages/wallet/src/redux/protocols/indirect-funding/state.ts

```diff
@@ -22,3 +22,7 @@ export function success(): Success {
 export function failure(): Failure {
   return { type: 'Failure' };
 }
+
+export function isTerminal(state: IndirectFundingState): state is Failure | Success {
+  return state.type === 'Failure' || state.type === 'Success';
```

Comment:
This is a case where "magic strings" can lead to an unsafe type guard. It's possible to change the type of `Failure` to `{ type: 'FAILURE' }`, which will render this type guard's implementation incorrect. The type system will not notice this, and there will be no compiler error.

Reply:
Since state is typed as `IndirectFundingState` the compiler will complain if the comparison `state.type === 'Failure'` doesn't include a valid type. So if we change `Failure` to`FAILURE` we get a compile error right away.

Reply:
Right, nice! It is still possible to have an issue, where one of the other `IndirectFundingState` has type `'Failure'`. But that's just an academic example and would be a much bigger problem. (Using an enum would eliminate that possibility.)